instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java

```diff
@@ -0,0 +1,351 @@
+                        span.hasName("SET")
+                            .hasKind(SpanKind.CLIENT)
+                            .hasParent(trace.getSpan(0))
+                            .hasAttributesSatisfyingExactly(
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1"),
+                                equalTo(
+                                    SemanticAttributes.NET_SOCK_PEER_NAME,
+                                    expectedHostAttributeValue),
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_PORT, port),
+                                equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                                equalTo(SemanticAttributes.DB_STATEMENT, "SET TESTSETKEY ?"))
+                            .hasEventsSatisfyingExactly(
+                                event -> event.hasName("redis.encode.start"),
+                                event -> event.hasName("redis.encode.end")),
+                    span ->
+                        span.hasName("callback")
+                            .hasKind(SpanKind.INTERNAL)
+                            .hasParent(trace.getSpan(0))));
+  }
+
+  @Test
+  void testGetCommandWithLambdaFunction()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    CompletableFuture<String> future = new CompletableFuture<>();
+
+    reactiveCommands
+        .get("TESTKEY")
+        .subscribe(
+            res -> {
+              assertThat(res).isEqualTo("TESTVAL");
+              future.complete(res);
+            });
+
+    assertThat(future.get(10, TimeUnit.SECONDS)).isEqualTo("TESTVAL");
+
+    getInstrumentationExtension()
+        .waitAndAssertTraces(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span ->
+                        span.hasName("GET")
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttributesSatisfyingExactly(
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1"),
+                                equalTo(
+                                    SemanticAttributes.NET_SOCK_PEER_NAME,
+                                    expectedHostAttributeValue),
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_PORT, port),
+                                equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                                equalTo(SemanticAttributes.DB_STATEMENT, "GET TESTKEY"))
+                            .hasEventsSatisfyingExactly(
+                                event -> event.hasName("redis.encode.start"),
+                                event -> event.hasName("redis.encode.end"))));
+  }
+
+  // to make sure instrumentation's chained completion stages won't interfere with user's, while
+  // still recording spans
+  @Test
+  void testGetNonExistentKeyCommand()
+      throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
I think it's perfectly reasonable to just `throws Exception` on all these. Reads a bit clearer to me.

Reply:
I tried updating these but encountered an issue:

```
warning: [InterruptedExceptionSwallowed] This method can throw InterruptedException but declares that it throws Exception/Throwable. This makes it difficult for callers to recognize the need to handle interruption properly.
  void testSetCommandWithSubscribeOnDefinedConsumer()
       ^
    (see https://errorprone.info/bugpattern/InterruptedExceptionSwallowed)
  Did you mean 'throws ExecutionException, InterruptedException, TimeoutException {'?
```

Reply:
nvm, i'll just suppress that